### ISSUE TITLE
Questions & comments 3

### ISSUE THREAD

**Issue by brucehohl** [andmarti1424/sc-im#145]
(1)  Does scim only load files with an .sc extension?
A file can be written as 'file.scim'.                                     
Attempts to open 'file.scim' resulted in:
'file.scim' is not a SC-IM compatible file.
It seems natural that a .scim file should be allowed.                     
              
(2)  Inserting double quotes into a cell freezes scim. Is there an       
escape character for inserting quotes as part of a text string?              
              
(3) Why does a cell allow text AND numeric content?
And, the results of entering text and numeric content
differ depending on the order entered.
              
    Order                                       Result    
    55 then <a                                   a    55.00
    55 then \a ('a' landed in column to right   a     55.00
    55 then >a ('a' is not displayed?)                55.00
                                                     165.00
              
    a then 55 (same as above)                    a    55.00
    a then 55 (same cell as expected)                a55.00
    a then 55 (same as above)                         55.00
                                                     165.00
              
(4) Can @if evaluate strings like for example:
@if(A1='a',B1,0)
If so, how?

(5)  While in normal mode when positioned in column B of attached file if        
[Enter] is pressed column B auto sizes ... it should not, right?

[questions-3.sc.zip](https://github.com/andmarti1424/sc-im/files/785316/questions-3.sc.zip)

**Comment by brucehohl** [comment]
Regarding Item (2):  Actually entering <". or >". or \". into a cell (even with other characters) causes the CPU to race and the program to freeze.  The same thing occurs in sc 7.16.

**Comment by andmarti1424** [comment]
1. you can open .csv files or other delimited files, if you passed them as argv..
sc extension was kept from old sc.
may be can be changed in future to .scim, as you suggest.
3. this comes from sc as well. about the different results depending on the order of things entered..it should not happen. i will have to check and fix them. 
4. have to check and remember how those functions work.
5. bug
2. not really possible right now, i believe. the later comment is bug as well.

Now I am working hard on issue #81. Its a lot of work and I might create another branch for it.
May be fix these issues and publish v0.5 release with this branch, since it already has bunch of fixes and improvents.

**Comment by brucehohl** [comment]
Issue #81 to add freeze rows & columns is a good feature.  I will check
back later after your work on 81 is complete.  This is a good project with
interesting possibilities.

**Comment by andmarti1424** [comment]
Regarding 5., the problem is:
CTRL is defined as
#define ctl(x)           ((x) & 0x1f)
then:
j = 106 = 1101010
0x1f = 31 = 0011111
1101010 & 0011111 = 0001010 = 10 = ENTER key..
And that is the problem.
I think ctl should be defined in another way, but I am not sure how..

**Comment by brucehohl** [comment]
Probably it was a bad idea for me to have put the five items together in on issue report.  If you want separate reports for some or all items let me know.

**Comment by brucehohl** [comment]
Regarding question 4:  The correct syntax is @if(@eqs(A1,"a"),B1,0)

**Comment by andmarti1424** [comment]
Will add that to doc. And will try to fix number 2, for next version.

**Comment by brucehohl** [comment]
More on original item (2) " Inserting double quotes into a cell freezes scim."
Correction entering as text the following ""."" freezes scim (and sc).  That is, a period between two sets of double quotes causes the freeze and cpu to race.  

More on original item (2) "Is there an escape character for inserting quotes as part of a text string?"
I think that sc originally used the backslash as an escape character:
In sc:
leftstring A1 = "\\"Quote\\""  displays  "Quote"
leftstring A2 = "Size = 55\\""  displays  Size = 55"

In scim:
leftstring A1 = "\\"Quote\\""  displays  \\"Quote\\"
leftstring A2 = "Size = 55\\""  displays  Size = 55\\"

scim seems to recognize backslash as a escape character for input but not for display.

**Comment by andmarti1424** [comment]
Just did a commit in "freeze" branch allowing double quotes to be entered like this:
\\"some text\"
Since it will be available not until next release, take note that its only a 3 line patch, in case you are interested in applying it to your source..

**Comment by andmarti1424** [comment]
I have also fixed the crash pointed in point 2. Commit is in the freeze branch also. Fix will be there for next release.

**Comment by andmarti1424** [comment]
Added your reply to number 4 to doc.

**Comment by andmarti1424** [comment]
The issue in number 3 ("..landed in column to right") also happens in sc!! Do not know if that was in purpose, but I do know like that behaviour. Just commented it in a new commit in freeze branch. Will be there as well for next version.

**Comment by andmarti1424** [comment]
I close this. The patches and fixed for these issues are on the default and only branch. The only thing left to fix is the number 5. Please fell free to open a new issue for that one.
Thanks!

**Comment by brucehohl** [comment]
OK.  Thanks for all that.  In the future I will keep it to one issue at at time :)  Looking forward to the next release.  BH
